pkg/analyzer/nodes/node_test.go

```diff
@@ -431,6 +434,47 @@ func Test_extractFilesFromDirectory(t *testing.T) {
 	}
 }
 
+func Test_extractFilesFromDirectory_Context(t *testing.T) {
+	defer goleak.VerifyNone(t)
```

Comment:
let's say we did not buffer the channel. Would this check fail?

Reply:
Yes, it would in some cases. We had a code that was using unbuffered channel for this: see https://github.com/stackrox/scanner/pull/1118#discussion_r1136321437
We decided to apply buffer of 1 message to avoid one additional `select` and make the code more compact and easier to read.